internal/cmd/login.go

```diff
@@ -231,8 +230,8 @@ func updateInfraConfig(client *api.Client, loginReq *api.LoginRequest, loginRes
 	return nil
 }
 
-func isNonInteractiveMode() bool {
-	return rootOptions.NonInteractive || os.Stdin == nil || !term.IsTerminal(int(os.Stdin.Fd()))
+func (o loginCmdOptions) isNonInteractiveMode() bool {
+	return o.NonInteractive || os.Stdin == nil || !term.IsTerminal(int(os.Stdin.Fd()))
```

Comment:
unfortunately this will end up getting replicated for each options type

Reply:
True. Right now there are only two commands that use this, and the second was not using this logic.

I think we will want to share this differently.

In the future I expect us to have `commonOptions` (or something like that) with flags for overriding what is in the client config. The interactive flag might be good on `commonOptions`.  I didn't want to make that change now so that the PR was more focused on just the changes to loading values from flags.

I think for now we could accept the boolean flag value to `isNonInteractiveMode`, so that this can be called from other places.

Reply:
yeah, or move o.NonInteractive out of this check, and require the caller to check it themselves

Reply:
Thinking about this more, the `os.Stdin == nil || !term.IsTerminal(int(os.Stdin.Fd())` should probably be the default set on the flag.

That way `--non-inactive=false` will work to override the stdin check. Otherwise there's no way to force it to be interactive when running without a terminal (which we might need for testing).

That has another plus, that now simply using the field from options is correct, and the contributor doesn't need to remember to call this function.

Reply:
fixed in the latest commit